samcli/local/docker/lambda_debug_settings.py

```diff
@@ -0,0 +1,213 @@
+"""
+Represents Lambda debug entrypoints.
+"""
+
+import json
+from collections import namedtuple
+
+from samcli.local.docker.lambda_image import Runtime
+
+
+class DebuggingNotSupported(Exception):
+    pass
+
+
+DebugSettings = namedtuple("DebugSettings", ["entrypoint", "debug_env_vars"])
+
+
+class LambdaDebugSettings:
+    @staticmethod
+    def get_debug_settings(debug_port, debug_args_list, runtime, options):
+        """
+        Get Debug settings based on the Runtime
+
+        Parameters
+        ----------
+        debug_port int
+            Port to open for debugging in the container
+        debug_args_list list(str)
+            Additional debug args
+        runtime str
+            Lambda Function runtime
+        options dict
+            Additonal options needed (i.e delve Path)
+
+        Returns
+        -------
+        tuple:DebugSettings (list, dict)
+            Tuple of debug entrypoint and debug env vars
+
+        """
+
+        entrypoint_mapping = {
+            Runtime.java8.value: DebugSettings(
+                entrypoint=["/usr/bin/java"]
+                + debug_args_list
+                + [
+                    "-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,quiet=y,address=" + str(debug_port),
+                    "-XX:MaxHeapSize=2834432k",
+                    "-XX:MaxMetaspaceSize=163840k",
+                    "-XX:ReservedCodeCacheSize=81920k",
+                    "-XX:+UseSerialGC",
+                    # "-Xshare:on", doesn't work in conjunction with the debug options
+                    "-XX:-TieredCompilation",
+                    "-Djava.net.preferIPv4Stack=true",
+                    "-jar",
+                    "/var/runtime/lib/LambdaJavaRTEntry-1.0.jar",
+                ],
+                debug_env_vars={},
+            ),
+            Runtime.java11.value: DebugSettings(
+                None,
+                debug_env_vars={
+                    "_JAVA_OPTIONS": f"-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,quiet=y,address=*:{debug_port} -XX:MaxHeapSize=2834432k -XX:MaxMetaspaceSize=163840k -XX:ReservedCodeCacheSize=81920k -XX:+UseSerialGC -XX:-TieredCompilation -Djava.net.preferIPv4Stack=true"
```

Comment:
Isn't this line too long for our `black` style check?

Reply:
If black can't figure out how to make it conform it will leave it. So it will still pass black (our CI checks for this as well)